stock_internal_transfer_transit/stock_internal_transfer.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 initOS GmbH & Co. KG (<http://www.initos.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+from openerp.osv import orm
+from openerp.netsvc import LocalService
+
+
+class StockInternalTransfer(orm.TransientModel):
+    _inherit = 'stock.internal.transfer'
+
+    def _prepare_picking(self, cr, uid, ids, context=None):
+        assert len(ids) == 1,\
+            'This function should only be used for a single id at a time.'
+
+        picking_data = super(StockInternalTransfer, self).\
+            _prepare_picking(cr, uid, ids, context=context)
+        wiz = self.browse(cr, uid, ids[0], context=context)
+
+        no_transit = context and \
+            context.get('stock_internal_transfer_no_transit', False)
+        if wiz.mode == 'force' or no_transit:
+            return picking_data
+
+        # update destination location to transfer using
+        # intermediate transit stock location
+        model_data_obj = self.pool['ir.model.data']
+        stock_loc_transit_id = model_data_obj.\
+            get_object_reference(cr, uid, 'stock_internal_transfer_transit',
+                                 'stock_location_transit')[1]
```

Comment:
Ok. I changed it like you suggested. This is an acceptable fix I think. 
To do it right every company would need to have its own transit location and goods should get transferred to the transit location belonging to the source company to reflect that goods are still owned by the sending company normally during transfer. 
But this would need more complex changes to create a proper transit location for every company during install of the module or on company creation.